apps/ec.c

```diff
@@ -174,12 +174,6 @@ int ec_main(int argc, char **argv)
         goto end;
     }
 
-    if (informat != FORMAT_ENGINE) {
-        in = bio_open_default(infile, 'r', informat);
```

Comment:
Hooooold on there.  `informat` may be any of these values:

https://github.com/openssl/openssl/blob/e34e91d7e575a2f69119601f2d34655cb6816148/apps/include/fmt.h#L20-L41

As you can see for yourself, some of them mean that the input file is (re?)opened in binary mode, on platforms where that matters...

Simply removing this may turn out to be a regression for non-Unix platforms (all other platforms do an implicit line ending conversion when in text mode, which is detrimental for binary input.

Mind you, a *different* mechanism to tell the command that the input is binary rather than text would be perfectly feasible.

Reply:
And the next lines also pass the informat..
  
  if (pubin)
        eckey = load_pubkey(infile, informat, 1, passin, e, "public key");
    else
        eckey = load_key(infile, informat, 1, passin, e, "private key");

Shouldnt these functions be the one handling this?

Reply:
@levitte The in is not used by the subsequent operations at all. So I am not sure how calling the bio_open_default could affect the subsequent operation of the store in the load_pubkey/load_key functions. Could you please explain?

Reply:
Could be, that we are missing call to `FMT_istext(format) ? BIO_FP_TEXT : 0` in load_key_certs_crls() but that is orthogonal to this issue.

Reply:
Somehow, I missed that `in` is complete unused.

So, er, don't mind me.  I must have been tired when I did that knee jerk...